packages/edge-functions/dev/node/main.ts

```diff
@@ -0,0 +1,302 @@
+   * instructs the handler to evaluate the functions and return their configs,
+   * which are then returned by this method.
+   */
+  private async getFunctionConfigs(denoPort: number, functions: Record<string, string>) {
+    const url = new URL(`http://${LOCAL_HOST}:${denoPort}`)
+    url.searchParams.set('functions', encodeURIComponent(JSON.stringify(functions)))
+
+    const res = await fetch(url, {
+      method: 'NETLIFYCONFIG',
+    })
+    const configs = (await res.json()) as Record<string, FunctionConfig>
+
+    return configs
+  }
+
+  /**
+   * Returns the list of edge functions that should run for a given request.
+   * It computes both the names of the edge functions that should run as well
+   * as the invocation metadata object that musto be included in the request.
+   */
+  private async getFunctionsForRequest(
+    req: Request,
+    functions: EdgeFunction[],
+    functionConfigs: Record<string, FunctionConfig>,
+  ) {
+    const url = new URL(req.url)
+    const declarations = mergeDeclarations(this.configDeclarations, functionConfigs, {}, [])
+    const { manifest } = generateManifest({
+      declarations,
+      userFunctionConfig: functionConfigs,
+      functions,
+    })
+    const functionNames: string[] = []
+    const routeIndexes: number[] = []
+    const routes = [...manifest.routes, ...manifest.post_cache_routes]
+
+    routes.forEach((route, index) => {
+      if (route.methods && route.methods.length !== 0 && !route.methods.includes(req.method)) {
+        return
+      }
+
+      const pattern = new RegExp(route.pattern)
+      if (!pattern.test(url.pathname)) {
+        return
+      }
+
+      const isExcludedForFunction = manifest.function_config[route.function]?.excluded_patterns?.some((pattern) =>
+        new RegExp(pattern).test(url.pathname),
+      )
+      if (isExcludedForFunction) {
+        return
+      }
+
+      const isExcludedForRoute = route.excluded_patterns.some((pattern) => new RegExp(pattern).test(url.pathname))
+      if (isExcludedForRoute) {
+        return
+      }
+
+      functionNames.push(route.function)
+      routeIndexes.push(index)
+    })
+
+    const invocationMetadata = {
+      function_config: manifest.function_config,
+      req_routes: routeIndexes,
+      routes: routes.map((route) => ({
+        function: route.function,
+        path: route.path,
+        pattern: route.pattern,
+      })),
+    }
+
+    return { functionNames, invocationMetadata }
+  }
+
+  /**
+   * Runs a request through the edge functions handler. The request may or may
+   * not match any edge functions: if it does, this method takes ownership of
+   * the request and returns the corresponding response; if it doesn't, the
+   * method returns `undefined`.
+   */
+  async handle(request: Request) {
+    if (request.headers.has(headers.Passthrough)) {
+      return
+    }
+
+    const functions = await find(this.directories)
```

Comment:
nit: `functions` and `functionsMap` contain _all_ functions but `functionNames` contains _matching_ functions. Could we rename either the former or the latter for clarity?